Core/TrackerResolver.swift

```diff
@@ -74,7 +74,7 @@ class TrackerResolver {
             return nil
         }
 
-        return DetectedTracker(url: trackerUrlString, knownTracker: tracker, entity: entity, blocked: blocked)
+        return DetectedTracker(url: trackerUrlString, knownTracker: tracker, entity: entity, blocked: blocked, pageUrl: pageUrlString)
```

Comment:
I've just realised that pageUrlString doesn't even come from the WKMessage like I thought it did, but is from message.webview.url, so good chance it doesn't even fix the race condition anyway 🤦‍♀️

Reply:
Existing code is
`guard let pageUrlStr = message.webView?.url?.absoluteString ?? dict[ContentBlockerKey.pageUrl] as? String else { return }`
So I'm not sure in what conditions we don't get ` dict[ContentBlockerKey.pageUrl]`, but I think we should just do 
`guard let pageUrlStr = dict[ContentBlockerKey.pageUrl] as? String else { return }`